R/dataset.R

```diff
@@ -428,27 +425,24 @@ webToAPIURL <- function (url) {
     return(absoluteURL(path, getOption("crunch.api")))
 }
 
-#' View a Dataset in the Web Application
+#' View a Crunch Object in the Web Application
 #'
 #' Convenience function that will use your system's "open" command to open
-#' a dataset in our web application in your default browser.
-#'
-#' Note that this function does not do anything on Windows.
+#' a Crunch object in our web application in your default browser.
 #'
-#' @param dataset a CrunchDataset
+#' @param x a Crunch Dataset or Variable
 #' @return Nothing; called for side effect of opening your web browser.
+#' @name webApp
+#' @importFrom utils browseURL
 #' @export
-webApp <- function (dataset) {
-    if (.Platform$OS.type == "unix") {
-        cmd <- ifelse(grepl("apple", R.version$platform), "open", "xdg-open")
-        url <- APIToWebURL(dataset)
-        system_call(cmd, url)
+webApp <- function(x) {
+    url <- try(APIToWebURL(x), silent = TRUE)
```

Comment:
Rather than try/catching here, you could make `APIToWebURL` fail with a nice message. Define a `APIToWebURL` method for `"ANY"` that does `halt("Web URL not available for objects of class ", class(x))`. cf. https://github.com/Crunch-io/rcrunch/blob/master/R/shoji.R#L69 or https://github.com/Crunch-io/rcrunch/blob/master/R/AllGenerics.R#L241-L244.